.github/workflows/cross-compile.yml

```diff
@@ -0,0 +1,115 @@
+on: workflow_dispatch
+
+env:
+  MAVEN_OPTS: -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
+
+jobs:
+  create-branch:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - run: echo "precompiled_branch=$(git branch --show-current)-precompiled-natives" >> $GITHUB_ENV
+      - run: echo "rev=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+      - name: Create branch for precompiled natives
+        run: git checkout -b ${{ env.precompiled_branch }} && git push --force origin HEAD
+    outputs:
+      precompiled_branch: ${{ env.precompiled_branch }}
+      base_rev: ${{ env.rev }}
+
+  compile-natives:
+    runs-on: ${{ matrix.os }}
+    needs: create-branch
+    strategy:
+      matrix:
+        include:
+          - target: linux_32
+            os: ubuntu-20.04
+            profile: 'x86'
+            packages: 'cmake g++-multilib'
+
+          - target: linux_64
+            os: ubuntu-20.04
+            profile: 'x86_64'
+            packages: 'cmake g++'
+
+          - target: linux_arm
+            os: ubuntu-20.04
+            profile: 'armhf'
+            packages: 'cmake g++-arm-linux-gnueabihf'
+
+          - target: linux_arm64
+            os: ubuntu-20.04
+            profile: 'aarch64'
+            packages: 'cmake g++-aarch64-linux-gnu'
+
+          - target: linux_ppc
+            os: ubuntu-20.04
+            profile: 'ppc64'
+            packages: 'cmake g++-powerpc64le-linux-gnu'
+
+          - target: windows_32
+            os: ubuntu-20.04
+            profile: 'mingw32'
+            packages: 'cmake g++-mingw-w64-i686'
+
+          - target: windows_64
+            os: ubuntu-20.04
+            profile: 'mingw64'
+            packages: 'cmake g++-mingw-w64-x86-64'
+
+          - target: windows_arm64
+            os: ubuntu-20.04
+            profile: 'mingwaarch64'
+            packages: 'cmake clang'
+
+          - target: osx_64
+            os: macos-10.15
+            profile: 'x86_64'
+
+          - target: osx_arm64
+            os: macos-10.15
+            profile: 'aarch64'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup APT packages
+        if: ${{ matrix.os == 'ubuntu-20.04' }}
+        run: sudo apt-get install --yes ${{ matrix.packages }}
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: 11
+          distribution: adopt
+      - name: Build with Maven
+        run: mvn -B clean install -P ${{ matrix.profile }}
+      - run: git config --global user.email "<>"
+      - run: git config --global user.name "Automated build on Github"
+      - run: git fetch && git checkout -t origin/${{ needs.create-branch.outputs.precompiled_branch }}
+      - run: git add src/main/resources-precompiled/**
+      - run: git commit --allow-empty -m "Precompiled natives (@${{ needs.create-branch.outputs.base_rev }}) for ${{ matrix.target }}"
```

Comment:
Possibly the resulting artifact is identical to the original - this results in an empty commit